### DISCUSSION TITLE
Risky rule options

### DISCUSSION THREAD

**Discussion by julienfalque** [PHP-CS-Fixer/PHP-CS-Fixer#7313] [Feature ideas]
Currently we are able to mark a rule as risky (meaning applying it might change code's behavior). Though this means every possible change that can be applied is considered risky.

Being able to mark only a single option of a non-risky rule would allow making rules that can apply non-risky changes by default and optionally apply risky changes. Current setup requires two separate rules.

Maybe a good use case would be #4652.

**Comment by SpacePossum** [comment]
:+1:  , this would also allow for a risky-option for `no_trailing_whitespace` to just do it everywhere :)

I run into this a bunch of times and would be really great to have this. (not sure about the product roadmap/prio's etc. of course)

**Comment by dmvdbrugge** [comment]
As long as the enable/disable of all individual options are still under the user's control, I'm all for this! (So not magically enabling all risky options because "allow risky is true")

**Comment by keradus** [comment]
I believe idea is valid, yet not planned as a focus nowadays.
Any implementation contribution will be great ;)
